lib/utils/flagship_tools.dart

```diff
@@ -0,0 +1,20 @@
+import '../flagship.dart';
+import 'logger/log_manager.dart';
+
+// length for the envId
+const int LengthId = 20;
+// pattern for envId
+const String xidPattren = "[0-9a-v]{20}";
```

Comment:
I think xid are only hexadecimal chars ([0-9a-f])

Reply:
To validate a base32 xid, expect a 20 chars long, all lowercase sequence of a to v letters and 0 to 9 numbers **([0-9a-v]{20}).**

https://github.com/rs/xid